sendgrid/base_interface.py

```diff
@@ -19,13 +20,18 @@ def __init__(self, auth, host, impersonate_subuser):
         :type impersonate_subuser: string
         :param host: base URL for API calls
         :type host: string
+        :param region: To determine the region which can only be 'global' or 'eu'
+        :type region: string
         """
         from . import __version__
         self.auth = auth
-        self.host = host
         self.impersonate_subuser = impersonate_subuser
         self.version = __version__
         self.useragent = 'sendgrid/{};python'.format(self.version)
+        if region is None:
```

Comment:
should the region is None check be inside the set_data_residency?  Or perhapse, should the default not be "none", but instead just be "global"?

Reply:
Here, I have not set the default value of region to be "global" because we are giving precedence to region in case both host and region values are present. If we set the default value to be global, the else part will never execute.